components/script/dom/response.rs

```diff
@@ -270,6 +219,71 @@ impl ResponseMethods<crate::DomTypeHolder> for Response {
         Ok(r)
     }
 
+    // https://fetch.spec.whatwg.org/#dom-response-json
+    #[allow(unsafe_code)]
+    fn CreateFromJson(
+        cx: JSContext,
+        global: &GlobalScope,
+        data: HandleValue,
+        init: &ResponseBinding::ResponseInit,
+        can_gc: CanGc,
+    ) -> Fallible<DomRoot<Response>> {
+        #[repr(C)]
+        struct VoidableString {
+            is_void: bool,
+            data: String,
+        }
+
+        let mut out_str = VoidableString {
```

Comment:
This name was based on what happens in Gecko, where the string has a void flag. I tried to do this with an Option::<String> but it ended up being pretty messy. Willing to rename this if desired.